src/main/java/org/microg/nlp/backend/nominatim/BackendService.java

```diff
@@ -304,4 +408,163 @@ String retrieveString() {
             return bos.toByteArray();
         }
     }
+
+    private List<Address> retrieveLocationFromCache(double latitude, double longitude, String locale) {
+        boolean useCache = PreferenceManager.getDefaultSharedPreferences(this).getBoolean(SettingsActivity.LOCATION_CACHE_ENABLED, true);
+
+        if (!useCache) {
+            return null;
+        }
+
+        Address addressFromCache = getResultFromCache(latitude, longitude, locale);
+        appendLog(TAG, "address retrieved from cache:" + addressFromCache);
+        if (addressFromCache == null) {
+            return null;
+        }
+        List<Address> addresses = new ArrayList<>();
+        addresses.add(addressFromCache);
+        return addresses;
+    }
+
+    private void storeAddressToCache(double latitude, double longitude, String locale, Address address) {
+
+        boolean useCache = PreferenceManager.getDefaultSharedPreferences(this).getBoolean(SettingsActivity.LOCATION_CACHE_ENABLED, true);
+
+        if (!useCache) {
+            return;
+        }
+
+        // Gets the data repository in write mode
+        SQLiteDatabase db = mDbHelper.getWritableDatabase();
+
+        // Create a new map of values, where column names are the keys
+        ContentValues values = new ContentValues();
+        values.put(LocationAddressCache.COLUMN_NAME_ADDRESS, getAddressAsBytes(address));
+        values.put(LocationAddressCache.COLUMN_NAME_LONGITUDE, longitude);
+        values.put(LocationAddressCache.COLUMN_NAME_LATITUDE, latitude);
+        values.put(LocationAddressCache.COLUMN_NAME_LOCALE, locale);
+        values.put(LocationAddressCache.COLUMN_NAME_CREATED, iso8601Format.format(new Date()));
```

Comment:
Please use unix timestamp in seconds or milliseconds for storing timestamps in database.

Reply:
fixed